pkg/util/keys.go

```diff
@@ -75,3 +120,23 @@ func GenerateObjectInfoString(obj runtime.Object) string {
 
 	return fmt.Sprintf("[%s] %s/%s", mObj.GetNamespace(), prefix, mObj.GetName())
 }
+
+func EmptyUnstructuredObjectFromKey(key Key) *unstructured.Unstructured {
+	obj := &unstructured.Unstructured{}
+
+	obj.SetGroupVersionKind(key.GVK)
+	obj.SetNamespace(key.NamespacedName.Namespace)
+	obj.SetName(key.NamespacedName.Name)
+
+	return obj
+}
+
+func KeyIsForPlacementDecision(key Key) bool {
+	if key.GVK.Group == v1alpha1.SchemeGroupVersion.Group &&
+		key.GVK.Version == v1alpha1.SchemeGroupVersion.Version &&
```

Comment:
I know that in-place upgrades are not in our current level of ambition, but even so: what goes wrong if you omit the version check?

Reply:
Addressed all comments written before (time-wise) this one.